manifests/init.pp

```diff
@@ -63,4 +63,23 @@
   $syslog_protocol    = 'udp',
   $syslog_port        = 514,
   $syslog_facility    = 'local7',
-) { }
+) {
+  
+  # The hackery is strong.  Not sure where else to hack this in.
+  if $::osfamily == "RedHat" {
+    file{'/etc/facter':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444'
+    }
+
+    file{'/etc/facter/facts.d':
+      ensure  => 'directory',
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444',
+      require => File['/etc/facter']
+    }
+  }
+}
```

Comment:
If you don't want to go that way, I'd get rid of the `::osfamily` conditional and wrap those suckers in `if ! defined(File[XXX) { }` blocks and assert that everyone is happy.